tools/source_updater/updater/main.py

```diff
@@ -0,0 +1,75 @@
+import click
+import glob
+import logging
+import os
+
+from airbyte_cdk.connector import BaseConnector
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
+from updater.catalog import ConfiguredCatalogAssembler, CatalogMerger
+from updater.config import Config
+from updater.handler import SourceUpdaterHandler
+from updater.source import SourceRepository
+
+_DEFAULT_CONFIG_FILE = "config.json"
+
+logger = logging.getLogger("main")
+
+
+def _assemble_configs(config_path: str):
+    """
+    :return: the main config and the set of the other configs
+    """
+    config_files = glob.glob(os.path.join(config_path, "*.json"))
+    main_config = None
+    other_configs = set()
+    for file in config_files:
+        if os.path.basename(file) == _DEFAULT_CONFIG_FILE:
+            main_config = Config(
+                "config",
+                dict(
+                    BaseConnector.read_config(
+                        config_path if config_path.endswith(_DEFAULT_CONFIG_FILE) else os.path.join(config_path, _DEFAULT_CONFIG_FILE)
+                    )
+                )
+            )
+        else:
+            other_configs.add(
+                Config(
+                    os.path.basename(file).replace(".json", ""),
+                    dict(
+                        BaseConnector.read_config(
+                            config_path if config_path.endswith(_DEFAULT_CONFIG_FILE) else os.path.join(config_path, _DEFAULT_CONFIG_FILE)
+                        )
+                    )
+                )
+            )
+
+    if not main_config:
+        raise ValueError(f"Could not find main config. Name should match {_DEFAULT_CONFIG_FILE}")
+    return main_config, other_configs
+
+
+@click.command()
+@click.option("--source", type=str, required=True, help="Name of the source. For example, 'source-jira'")
+@click.option("--manifest", type=str, required=True, help="Path to the new yaml manifest file")
+@click.option("--debug", is_flag=True, default=False, help="Enable debug logs")
+def main(source: str, manifest: str, debug: bool) -> None:
+    source_name = source
+    manifest_path = manifest
+    if debug:
+        logging.basicConfig(level=logging.DEBUG, force=True)
+
+    logger.info("Starting the update...")
+    repo = SourceRepository()
```

Comment:
nit: doesn't look like we use the repo anywhere until we pass it to `SourceUpdaterHandler`. We could just instantiate it inside there w/o the need for another variable.

Also, since all the parameters that we pass to `SourceUpdaterHandler` are the defaults, we can just get rid of all the inputs and initialize them in the `__init__` constructor. wdyt?

Reply:
I added them as parameters in the init so that I can just pass a mock during tests. Default arguments does not seem advisable as it would be shared between all instances. Else, we could make the param optional in the __init__ params and if None instantiate it. I don't like this too much because it means that if the interface of `SourceRepository` changes (the __init__ method specifically here), `SourceUpdaterHandler` might change as well. However, I agree that `SourceUpdaterHandler()` seems cleaner than `SourceUpdaterHandler(SourceRepository(), CatalogMerger(ConfiguredCatalogAssembler()))`. Let me know what you prefer!

Reply:
>Default arguments does not seem advisable as it would be shared between all instances. Else, we could make the param optional in the init params and if None instantiate it.

Sorry yeah i could have been more clear, not default construct args, but just the __init__ will create a new instance of `SourceRepository()`. In that case it wouldn't be shared between instances of `SourceUpdaterHandler()` because each new creation is its own repo/merger. I would've thought? Fair point about the testing, happy to leave as is and only make the change to `SourceUpdaterHandler(SourceRepository(), CatalogMerger(ConfiguredCatalogAssembler()))`